addons/mass_mailing/models/mailing.py

```diff
@@ -240,10 +240,14 @@ def create(self, values):
 
         # check for duplicate name in db
         base_name = values['name']
-        duplicate_count = 1
-        while self.env['utm.source'].search_count([('name', '=', values['name'])]) > 0:
-            duplicate_count += 1
-            values['name'] = base_name + " [%s]" % duplicate_count
+        match = re.match(".*( \[[0-9]+\])$", base_name)
+        if match:
+            remove_char_count = len(match.groups()[0]) * -1
+            base_name = base_name[:remove_char_count]
+
+        duplicate_count = self.env['utm.source'].search_count([('name', '=like', base_name + "%")])
+        if duplicate_count > 0:
+            values['name'] = base_name + " [%s]" % (duplicate_count+1)
```

Comment:
same here, fixup it into the first commit

Reply:
commits squashed

Reply:
thanks for your review